middleware/service/unittest/source/manager/test_manager.cpp

```diff
@@ -1169,5 +1169,53 @@ namespace casual
          }
       }
 
+      TEST( service_manager, callee_dies_during_service_call__expect_metric)
+      {
+         common::unittest::Trace trace;
+
+         auto domain = local::domain();
+
+         service::unittest::advertise( { "a"});
+
+         {
+            auto service = common::service::Lookup{ "a"}();
+            EXPECT_TRUE( service.service.name == "a");
+            EXPECT_TRUE( service.process == common::process::handle());
+            EXPECT_TRUE( service.state == decltype( service.state)::idle);
+         }
+
+         // subscribe to metric event
+         common::message::event::service::Calls event;
+         common::event::subscribe( common::process::handle(), { event.type()});
+
+         // we "die"
+         {
+            common::message::event::process::Exit message;
+            message.state.pid = common::process::id();
+            message.state.reason = decltype( message.state.reason)::core;
+
+            common::communication::device::blocking::send( 
+               common::communication::instance::outbound::service::manager::device(),
+               message);
+         }
+
+         // we have to resubscribe since DM thinks we died
+         common::event::subscribe( common::process::handle(), { event.type()});
+
```

Comment:
This seems like it could be a timing glitch?
And why would DM think we've died?

Reply:
Ahh, my bad. It's the SM that handles all service metric. 
Hmm. You could subscripe to the event with a dedicated ipc-device, with a pid != our pid.